### PR TITLE
Add test for duplicate `PYTHONPATH` in `modextrapaths`

### PR DESCRIPTION
**Pull request by Flamefire**
And similar test for `sanity_check_paths`.

Intended to avoid issues fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/13385

### CODE REVIEW

**Comment by Flamefire** [comment]
@boegel Anything missing here? This is a CI test change only so shouldn't cause any regressions for users (but actually helps avoiding them)

**Comment by Flamefire** [comment]
Retargeteed 5.0x

**Comment by boegel** [comment]
@Flamefire Sync with `5.0.x` now that #22061 is merged?

**Comment by Flamefire** [comment]
@boegel Rebased this again and extended to checking the resolved values as well as one instance used `/lib/python%s...' % local_pyshortver` which will now be caught too

Can this be merged before new ECs with those paths get introduced?

**Comment by boegel** [comment]
@Flamefire Needs more love (again), sorry...

**Comment by Flamefire** [comment]
Rebased. Trivial merge conflict after a cleanup of a condition

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -1330,30 +1330,54 @@ def test_pr_python_packages(self):
...
+                'lib/python%(pyshortver)s/site-packages',
+            ]
+            with ec.allow_unresolved_templates():
+                default_dirs += [ec.resolve_template(d) for d in default_dirs]
+            extra_python_path = ec.get('modextrapaths', dict()).get('PYTHONPATH')
+            failing_checks_ec.extend(f"modextrapaths should not contain {d} (automatically added by EasyBlock)"
+                                     for d in default_dirs if d in extra_python_path)
+            if easyblock == 'PythonBundle' or easyblock.endswith('PythonPackage'):
+                sanity_check_dirs = ec.get('sanity_check_paths', dict()).get('dirs') or []
+                default_dirs = (
```
Comment (akesandgren) [inline test/easyconfigs/easyconfigs.py]: Should this  use [ and not ( ??

Comment (Flamefire) [reply test/easyconfigs/easyconfigs.py]: The general guideline I follow is: If the list doesn't need to be modified it will be a tuple

Not sure if this is also faster (due to less capabilities to cater for) but at least it avoids accidental changes where none are intended

Comment (akesandgren) [reply test/easyconfigs/easyconfigs.py]: But you are modifying it in the with ec.allow... for loop, line 1376

Comment (Flamefire) [reply test/easyconfigs/easyconfigs.py]: Good catch, not sure how I missed that as it would always fail

**Review by akesandgren — DISMISSED** [review]
LGTM

**Comment by akesandgren** [comment]
@boegel Please check this out. It looks good and clean to me...

**Comment by akesandgren** [comment]
@Flamefire Hmm... I just caused this to have a conflict when merging 21362

Please adjust it

**Comment by Flamefire** [comment]
Rebased and fixed
